### PR TITLE
DolphinQt: small input mapping code cleanups and fixes

### PR DESCRIPTION
**Pull request by ligfx**

### CODE REVIEW

**Comment by ligfx** [comment]
@spycrab @Starsam80

File: Source/Core/DolphinQt2/Config/Mapping/MappingButton.cpp [hunk 1]
```diff
@@ -89,25 +91,18 @@ void MappingButton::Update()
...
 {
-  return !m_block ? QPushButton::event(event) : true;
+  const QEvent::Type event_type = event->type();
+  // Returning 'true' means "yes, this event has been handled, don't propagate it to parent
+  // widgets".
+  if (m_block.IsSet() &&
+      (event_type == QEvent::KeyPress || event_type == QEvent::KeyRelease ||
+       event_type == QEvent::MouseButtonPress || event_type == QEvent::MouseButtonRelease ||
+       event_type == QEvent::MouseButtonDblClick))
+    return true;
```
Comment (lioncash) [inline Source/Core/DolphinQt2/Config/Mapping/MappingButton.cpp]: This should be braced since the condition body spans more than one line.

Comment (ligfx) [inline Source/Core/DolphinQt2/Config/Mapping/MappingButton.cpp]: Sure, makes sense.

**Comment by shuffle2** [comment]
hmmm the ugly detached thread strikes again....

**Pull request merged**